### ISSUE TITLE
[BUG] Variables from dead else-if branch are hoisted to incorrect location

### ISSUE THREAD

**Issue by shrinktofit** [rollup/rollup#3780]
https://github.com/rollup/rollup/pull/3752 did not consider such case:

```js
if (globalThis.notDead) {
    console.log('true-branch');
} else if (false) {
    var w; // `const` works fine here
    console.log('else-if-branch');
} else {
    console.log('else-branch');
}
```

which incorrectly outputs:
```js
if (globalThis.notDead) {
    console.log('true-branch');
} else var w; {
    console.log('else-branch');
}
```

Repl(since `2.26.7`): https://rollupjs.org/repl/?version=2.26.7&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmlmJTIwKGdsb2JhbFRoaXMubm90RGVhZCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwY29uc29sZS5sb2coJ3RydWUtYnJhbmNoJyklM0IlNUNuJTdEJTIwZWxzZSUyMGlmJTIwKGZhbHNlKSUyMCU3QiU1Q24lMjAlMjAlMjAlMjB2YXIlMjB3JTNCJTVDbiU1Q3QlMjAlMjBjb25zb2xlLmxvZygnZWxzZS1pZi1icmFuY2gnKSUzQiU1Q24lN0QlMjBlbHNlJTIwJTdCJTVDbiUyMCUyMCUyMCUyMGNvbnNvbGUubG9nKCdlbHNlLWJyYW5jaCcpJTNCJTVDbiU3RCUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmVzJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=

**Comment by lukastaegert** [comment]
The "incorrect" here is that `w` is unused, otherwise the code is valid and working?

**Comment by lukastaegert** [comment]
Thanks for spotting, fix at #3782.
